### ISSUE TITLE
Non-linear node fails for single multivariate input

### ISSUE THREAD

**Issue by bartvanerp** [biaslab/ReactiveMP.jl#328]
The non-linear node fails when it is provided with a single multivariate input, being mapped to a univariate output, e.g. 
```julia
@model function model_test()

    z ~ MvNormalMeanCovariance(zeros(2), I)
    x ~ foo(z)
    y = datavar(Float64)
    y ~ NormalMeanVariance(x, 1.0)
    
end
```

It produces the error `DimensionMismatch: derivative(f, x) expects that x is a real number. Perhaps you meant gradient(f, x)?`

**Comment by albertpod** [comment]
@bartvanerp I suppose this problem occurs only for Linearization, right?

**Comment by bartvanerp** [comment]
I haven't checked it for other unscented or CVI, but it fails when doing ForwardDiff magic, so I suppose you are right.

**Comment by bartvanerp** [comment]
Solved by #330
